src/main/java/ca/corefacility/bioinformatics/irida/plugin/staramr/StarAMRUpdater.java

```diff
@@ -0,0 +1,284 @@
+package ca.corefacility.bioinformatics.irida.plugin.staramr;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Joiner;
+import com.google.common.base.Splitter;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+
+import ca.corefacility.bioinformatics.irida.exceptions.IridaWorkflowNotFoundException;
+import ca.corefacility.bioinformatics.irida.exceptions.PostProcessingException;
+import ca.corefacility.bioinformatics.irida.model.sample.MetadataTemplateField;
+import ca.corefacility.bioinformatics.irida.model.sample.Sample;
+import ca.corefacility.bioinformatics.irida.model.sample.metadata.MetadataEntry;
+import ca.corefacility.bioinformatics.irida.model.sample.metadata.PipelineProvidedMetadataEntry;
+import ca.corefacility.bioinformatics.irida.model.workflow.IridaWorkflow;
+import ca.corefacility.bioinformatics.irida.model.workflow.analysis.AnalysisOutputFile;
+import ca.corefacility.bioinformatics.irida.model.workflow.analysis.type.AnalysisType;
+import ca.corefacility.bioinformatics.irida.model.workflow.submission.AnalysisSubmission;
+import ca.corefacility.bioinformatics.irida.pipeline.results.updater.AnalysisSampleUpdater;
+import ca.corefacility.bioinformatics.irida.service.sample.MetadataTemplateService;
+import ca.corefacility.bioinformatics.irida.service.sample.SampleService;
+import ca.corefacility.bioinformatics.irida.service.workflow.IridaWorkflowsService;
+
+/**
+ * {@link AnalysisSampleUpdater} for AMR detection results to be written to
+ * metadata of {@link Sample}s.
+ */
+public class StarAMRUpdater implements AnalysisSampleUpdater {
+	private static final Logger logger = LoggerFactory.getLogger(StarAMRUpdater.class);
+	private static final String STARAMR_SUMMARY = "staramr-summary.tsv";
+
+	private static final Splitter SPLITTER = Splitter.on('\t');
+
+
+	private static final String METADATA_STARAMR_GENE = "staramr/gene";
+	private static final String METADATA_STARAMR_DRUG_CLASS = "staramr/drug-class";
+    private static final String METADATA_STARAMR_QUALITY_MODULE_CLASS = "staramr/quality-module-class";
```

Comment:
Could you remove the `-class` end for all these variables, except for `drug-class` (which is referring to the class of antimicrobial drugs)?